setup1-contributor/src/commands/contribute.rs

```diff
@@ -222,6 +226,34 @@ impl Contribute {
             delay_for(Duration::seconds(DELAY_WAIT_FOR_PIPELINE_SECS).to_std()?).await;
         }
 
+        let cloned = self.clone();
+        //TODO: this is broken, probably because there is some
+        // blocking code in one of the other tasks.
+        //
+        // let join_handle: JoinHandle<anyhow::Result<()>> = tokio::task::spawn(async move {
+        //     let auth_rng = &mut rand::rngs::OsRng;
+        //     loop {
+        //         println!("Performing heartbeat.");
+        //         tracing::info!("Performing heartbeat.");
+        //         cloned.heartbeat(auth_rng).await?;
+        //         delay_for(heartbeat_poll_duration).await;
+        //     }
+        // });
+        // futures.push(join_handle);
+
+        std::thread::spawn(move || {
+            let auth_rng = &mut rand::rngs::OsRng;
+            let mut runtime = tokio::runtime::Runtime::new().unwrap();
+            loop {
+                println!("Performing heartbeat.");
```

Comment:
is it occasionally left here or intensionally? It's a bit hard to manage such logs in case you want to hide them

Reply:
also forgot to remove, thanks!

Reply:
I was printing in both places out of desperation :sweat_smile: but got it to work when I switched from a task to a dedicated thread.